server/src/instant/reactive/session.clj

```diff
@@ -476,7 +476,7 @@
                                     ;; If false, then canceling the queries let
                                     ;; the future complete before we could cancel it
                                     :future-cancel-result cancel-res}}))
-              (ex/throw-operation-timeout! :handle-receive handle-receive-timeout-ms)))
+              (ex/throw-operation-timeout! :handle-receive handle-receive-timeout-ms :in-progress-stmts in-progress-stmts)))
```

Comment:
The `in-progress-stmts` is just a list closures that let use cancel database queries for an event that times out.

Created a PR here that should do what you want: https://github.com/instantdb/instant/pull/523

Reply:
@dwwoelfel perfect ty I'll close this!